src/particles/CollectLost.cpp

```diff
@@ -55,8 +55,26 @@ namespace impactx
         BL_PROFILE("impactX::collect_lost_particles");
 
         using SrcData = ImpactXParticleContainer::ParticleTileType::ConstParticleTileDataType;
-
         ImpactXParticleContainer& dest = *source.GetLostParticleContainer();
+
+        // Check destination has the same attributes as source + "s_lost"
+        for (auto & name : source.GetRealSoANames())
+        {
+            if (!dest.HasRealComp(name)) {
+                dest.AddRealComp(name);
+            }
```

Comment:
@atmyers question for you: we use `amrex::filterAndTransformParticles` below.
I am a bit afraid that when the order of attributes added in `dest` is not 100% the same as in `source` that this gets confused.

In particular, the attributes in `source` are something like:
- x,y,t,px,py,pt,I,H
and in `dest`
- x,y,t,px,py,pt,s,I,H

`s`, `I` and `H` are runtime components, all are `Real`.

Will that lead to broken/mismatched or partial copies from `source` to `dest`?

Reply:
I tried to add a few asserts, but maybe need to generalize the matching a bit to simplify complex bookkeeping.